### PR TITLE
New DCR Sample, including settings and Portal config

### PR DESCRIPTION
**Pull request by rewsmith**
## Description
A new Operator sample which includes Portal DCR Settings, an API, Security Policy and Portal Catalog entry

## Motivation and Context
No samples for DCR currently

## Test Coverage For This Change

- Applied the manifest
- Created an OAuth Client in Portal (using Keycloak DCR under covers)
- Requested an access token from keycloak
- Used access token in call to API

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: config/samples/portal/dcr.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  protocol: http
+  active: true
+  proxy:
+    target_url: http://httpbin.org
+    listen_path: /httpbin
+    strip_listen_path: true
+  strip_auth_data: true
+  jwt_signing_method: rsa
+  jwt_source: aHR0cDovL2VjMi0xMy00MS0xNjUtMTUuZXUtd2VzdC0yLmNvbXB1dGUuYW1hem9uYXdzLmNvbTo4MTgwL2F1dGgvcmVhbG1zL21hc3Rlci9wcm90b2NvbC9vcGVuaWQtY29ubmVjdC9jZXJ0cw==
+  jwt_identity_base_field: sub
```
Comment (asoorm) [inline config/samples/portal/dcr.yaml:18]: I would use clientID here as opposed to sub. Usually you would want to rate-limit by the client, not the logged in user which the sub represents in OIDC.

Comment (rewsmith) [reply config/samples/portal/dcr.yaml:18]: Fair point.  Will correct

**Review by andrei-tyk — APPROVED** [review]

**Comment by rewsmith** [comment]
Please merge :-)

**Comment by buraksekili** [comment]
Thank you @rewsmith

**Pull request merged**